### ISSUE TITLE
3D Step or STL file available of the PCB and components to help in a case design?

### ISSUE THREAD

**Issue by chunkysteveo** [lizardsystems/yawl-controller#7]
Hi!

I have ordered parts and PCB's for 5 units, and built one with great success! It's a really neat board and very compact!

Just looking at designing my own case to 3D print - do you have the 3D render seen in STEP file format, or STL format? That would help in aligning all the walls and holes etc.

Thanks in advance.

**Comment by vzagorovskiy** [comment]
You may use  AK-S-80 case datasheet

- DWG https://github.com/lizardsystems/yawl-controller/blob/main/datasheets/AK-S-80.dwg
- DXF https://github.com/lizardsystems/yawl-controller/blob/main/datasheets/AK-S-80.dxf
- PDF https://github.com/lizardsystems/yawl-controller/blob/main/datasheets/AK-S-80.pdf

**Comment by chunkysteveo** [comment]
Never mind, I have the PCB layout DXF that I can work around, it's just easier to place holes and support pillars when you have the 3D file within Fusion360.

Thanks for your help though.

**Comment by numindast** [comment]
> Never mind, I have the PCB layout DXF that I can work around, it's just easier to place holes and support pillars when you have the 3D file within Fusion360.

Would you mind sharing your case design at some point?

**Comment by chunkysteveo** [comment]
Yeah sure thing! It will be very custom for my needs, but happy to push it to Thingiverse and this repo if he wants it 👍🏻

**Comment by numindast** [comment]
> Yeah sure thing! It will be very custom for my needs, but happy to push it to Thingiverse and this repo if he wants it 👍🏻

I'm terrible (yet) at 3D modeling so having a starting point is really helpful :) Even if custom to your needs it'll hopefully help me learn f360 better too. Thx!!

**Comment by chunkysteveo** [comment]
Hey @numindast  - I kind of ran out of time designing a box for the YAWL for this year, so adapted an existing schematic design that I use for other projects. This fits the controller well, no antenna, and 2 5mm holes for wires. May help you if you need a box urgently?! I've got various versions that are for 2 wire, 3 wire, antenna hole etc, but this is the basic one if it helps... 
[box_complete_2_holes_no_ant.zip](https://github.com/lizardsystems/yawl-controller/files/10084308/box_complete_2_holes_no_ant.zip)

**Comment by vzagorovskiy** [comment]
@chunkysteveo Could you give me permission to  add this file to the repository please?

**Comment by chunkysteveo** [comment]
Of course!! It's technically from this design on Thingiverse - https://www.thingiverse.com/thing:1680291 which asks for attribution. My design is a modified version of his SCAD file.

Here's a 2 hole and antenna version too...
[box_complete_2_holes_ant.zip](https://github.com/lizardsystems/yawl-controller/files/10085266/box_complete_2_holes_ant.zip)

![box_complete_2_holes_ant](https://user-images.githubusercontent.com/14663002/203805669-f7c35813-9296-46b3-8861-d7b214c2c1ca.png)

Feel free to add the files officially to the repo.

The screw holes are 3.4mm diameter, which allows me to get an M3 screw through the lid, and melt an M3 threaded insert into the body.

**Comment by chunkysteveo** [comment]
![20221124_141858](https://user-images.githubusercontent.com/14663002/203806490-191350f9-8534-4d18-b864-184726b7ed42.jpg)

Here's one in the flesh.... rain proof for Christmas lights outside.

**Comment by chunkysteveo** [comment]
If you download the SCAD file from https://www.thingiverse.com/thing:1680291 and you know OpenSCAD - you can manipulate the base layout to your hearts content! Maybe some little spaces to secure the PCB down to the base of the box would be useful, but it's snug enough!

**Comment by MandicReally** [comment]
Old issue but asking again, any chance you could release a STEP file or similar CAD file? The supplied case design is… really suboptimal imo. I will gladly design a case (and supply the files to include in the repo if you want). I dug theory the files but I’m not find a board DXF to produce my own design from. I got the PCBs from an existing PCBWay listing for the project. Am I just missing the DXF? It’s referenced but nowhere to be found that I can see. 

![Image](https://github.com/user-attachments/assets/8714f693-d670-4355-80aa-e3df261f9480)

**Comment by chunkysteveo** [comment]
My case design gets the job done 😂

What's the huge hole at the top used for, if you don't mind me asking?

**Comment by MandicReally** [comment]
> My case design gets the job done 😂
> 
> What's the huge hole at the top used for, if you don't mind me asking?

I’d say your case mostly gets the job done, but only after I removed the bottom-side components (IR receiver and button) and added a hole in the lid to clear the fuse. Even then, the PCB just floats inside. It’s a decent workaround considering there's no proper model, but I’d really like to design a purpose-built enclosure that actually supports the full board and mounts cleanly.

I have several of these from PCBWay for upcoming projects, and right now I’m just taping them into whatever box works. For example, I’m building a custom PC case that uses one as an RGB controller, and it would be great to integrate it properly instead of hacking something together.

I’m happy to contribute a refined case design to the repo, but I really need a DXF or board outline to work from. Reverse-engineering the board by hand feels like unnecessary effort when the file clearly existed before. If there’s any way to re-share it, that would be a huge help.

Oh and the hole is to allow a clear path to the wifi antenna. Necessary? Almost certainly not, but it cannot hurt. I'm not using these in adverse weather conditions so I don't need a sealed up box. I'd rather a little airflow and clear wifi signal path.
